### PR TITLE
UserAuth: Change to django users auth

### PR DESCRIPTION
**Pull request by ronyyosef**
Relevant issue: #91 #84 
Main Changes:
1. Change the House model to be with a one-to-one relationship with the Django user model
2. Change the login page to be Django user login
3. Add House after the first Login with the Django user
4. Logout
5. fix the create_mock_data to the new data models

### CODE REVIEW

**Comment by ShellyGolden** [comment]
Can you add screenshots? 🙏🏼
Ill take a look at the PR later

**Comment by ronyyosef** [comment]
image
image
image
image

**Comment by adiabramovitch** [comment]
This is a big PR with multiple functionalities. 
Can you split it into separate PRs according to the logic? 
It will be easier to review that way. 
Please add a description of what you intend to implement in each PR.

**Comment by ronyyosef** [comment]
> This is a big PR with multiple functionalities. Can you split it into separate PRs according to the logic? It will be easier to review that way. Please add a description of what you intend to implement in each PR.

Its abit problem here to split it into different PRs
After i changed the house model most of are project was broke
I fixes everything here tests, login, script for mock

I think it will be a good idea to have a meeting togther on this PR

**Comment by ShellyGolden** [comment]
Can you change the title of the PR?

File: Xpense/urls.py [hunk 1]
```diff
@@ -18,7 +18,8 @@
 
 urlpatterns = [
     path('admin/', admin.site.urls),
+    path('accounts/', include("accounts.urls"), name="accounts"),
+    path('accounts/', include("django.contrib.auth.urls")),
```
Comment (ShellyGolden) [inline Xpense/urls.py:22]: lines 21 and 22 can be one line?

File: expenses/models.py [hunk 2]
```diff
@@ -1,8 +1,9 @@
+from django.core.validators import MinValueValidator
 from django.db import models
 from django.db.models import Avg, IntegerField, QuerySet
 from django.utils import timezone
+
```
Comment (ShellyGolden) [inline expenses/models.py:5]: You can delete this empty line

File: house/migrations/0001_initial.py [hunk 3]
```diff
@@ -1,16 +1,17 @@
-# Generated by Django 4.0.3 on 2022-04-06 21:09
+# Generated by Django 4.0.3 on 2022-04-25 09:23
```
Comment (ShellyGolden) [inline house/migrations/0001_initial.py:2]: you can delete this line

**Comment by ronyyosef** [comment]
I will squash the commits close to the  merge

File: house/tests/house_tests.py [hunk 4]
```diff
@@ -1,10 +1,11 @@
 import pytest
+
```
Comment (ShellyGolden) [inline house/tests/house_tests.py:2]: empty line to delete

File: house/tests/house_tests.py [hunk 5]
```diff
@@ -137,14 +142,13 @@ def form_data_filter_tests():
...
 class TestMyHouseViews:
-    def test_house_view_function_200(self, db, generate_get_request):
-        HouseFactory().save()
-        house = House.objects.all()[0]
-        try:
-            render = house_view(generate_get_request, house.house_id)
-            assert render.status_code == 200, "Status code is not 200"
-        except Exception:
-            assert False, "Error in house_view function"
+    # def test_house_view_function_200(self, db, generate_get_request):
```
Comment (ShellyGolden) [inline house/tests/house_tests.py:145]: you can delete the lines that in comment

File: house/views.py [hunk 6]
```diff
@@ -1,11 +1,13 @@
...
-from house.constants import HOME_PAGE_ROUTE, GLOBAL_PAGE_ROUTE, GLOBAL_PAGE_CITY_DROPDOWN_ROUTE, LOGIN_PAGE_ROUTE
-from .forms import HouseForm, HouseIDForm
-from .helpers import _filter_houses_by_form
-from expenses.models import Expenses
-from django.http import HttpResponseRedirect
+from django.contrib.auth.decorators import login_required
 from django.core.exceptions import ValidationError
+from django.http import HttpResponseRedirect
+from django.shortcuts import render
+
```
Comment (ShellyGolden) [inline house/views.py:5]: empty line to delete

**Review by ShellyGolden — COMMENTED** [review]
It looks good 
I left a comment where you left empty lines to delete
Can you delete the empty files that they are not needed?

File: house/models.py [hunk 7]
```diff
@@ -46,8 +46,7 @@ class Job(models.TextChoices):
 
 
 class House(models.Model):
-    house_id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-    name = models.CharField(max_length=50)
```
Comment (ronyyosef) [inline house/models.py:50]: return

**Comment by LiorKGOW** [comment]
SInce it's in draft mode, i'll remove the ready for review label 👍

**Pull request closed without merging**